cwltool/job.py

```diff
@@ -94,18 +94,20 @@ def relink_initialworkdir(
             host_outdir_tgt = os.path.join(
                 host_outdir, vol.target[len(container_outdir) + 1 :]
             )
+            mode = (
+                os.stat(host_outdir_tgt).st_mode
+                | stat.S_IWUSR
+                | stat.S_IWGRP
+                | stat.S_IWOTH
+            )
             if os.path.islink(host_outdir_tgt) or os.path.isfile(host_outdir_tgt):
-                try:
-                    os.remove(host_outdir_tgt)
-                except PermissionError:
-                    pass
+                os.chmod(host_outdir_tgt, mode)
```

Comment:
https://github.com/common-workflow-language/cwltool/runs/7851533013?check_suite_focus=true#step:8:146

```
 File "/home/runner/work/cwltool/cwltool/cwltool/job.py", line 104, in relink_initialworkdir
      os.chmod(host_outdir_tgt, mode)
  PermissionError: [Errno 1] Operation not permitted
```

Reply:
Yes, I'm still working on the precise change necessary which both makes my code work and which satisfies your CI. Part of the problem is that your CI still doesn't work on my local machine. For example, if I do a fresh clone of your main branch (i.e. not my fork), and run pytest in a fresh conda environment, I get:

```
    def test_single_process_subwf_subwf_inline_step() -> None:
        """Test --single-process on an inline sub-sub-workflow step."""
        err_code, stdout, stderr = get_main_output(
            [
                "--single-process",
                "step1/stepX/stepY",
                get_data("tests/subgraph/count-lines17-wf.cwl.json"),
                get_data("tests/wf/wc-job.json"),
            ]
        )
        assert err_code == 0
>       assert (
            json.loads(stdout)["output"]["checksum"]
            == "sha1$3596ea087bfdaf52380eae441077572ed289d657"
        )
E       AssertionError: assert 'sha1$9aefd46...8d192af7bcac4' == 'sha1$3596ea0...7572ed289d657'
E         - sha1$3596ea087bfdaf52380eae441077572ed289d657
E         + sha1$9aefd466231cf30b4e9ae5234d58d192af7bcac4
```

I'm not really sure what's happening here. Maybe it's because I'm running on a Macbook?

Reply:
Ah, mac do have different `wc` command outputs; I'll make a PR to use a docker container for those tests.